lib/src/log.dart

```diff
@@ -1,28 +1,63 @@
 import 'dart:developer' as developer;
 
+import 'package:fast_immutable_collections/fast_immutable_collections.dart';
 import 'package:flutter/foundation.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:lichess_mobile/src/model/settings/log_preferences.dart';
 import 'package:logging/logging.dart';
 
 const _loggersToShowInTerminal = {'HttpClient', 'Socket', 'EvaluationService'};
 
-/// Setup logging
-void setupLogging() {
-  if (kDebugMode) {
-    Logger.root.level = Level.FINE;
+typedef LogState = ({IList<LogRecord> logs});
+
+/// Provides an instance of [AppLogStorage] using Riverpod.
+final appLogStorageProvider = NotifierProvider<AppLogStorage, LogState>(
+  AppLogStorage.new,
+  name: 'AppLogStorageProvider',
+);
+
+/// Manages log entries created via [Logger] instances
+///
+/// Currently, simply saves log entries in memory, so they do not persists across app restarts.
+class AppLogStorage extends Notifier<LogState> {
+  @override
+  LogState build() => (logs: const IList.empty());
+
+  void setup() {
+    ref.listen(logPreferencesProvider.select((prefs) => prefs.level), (prev, next) {
+      if (next != prev) {
+        Logger.root.level = next;
+      }
+    }, fireImmediately: true);
+
     Logger.root.onRecord.listen((record) {
-      developer.log(
-        record.message,
-        time: record.time,
-        name: record.loggerName,
-        level: record.level.value,
-        error: record.error,
-        stackTrace: record.stackTrace,
-      );
-
-      if (_loggersToShowInTerminal.contains(record.loggerName) && record.level >= Level.FINE) {
-        debugPrint('[${record.loggerName}] ${record.message}');
+      if (kDebugMode) {
+        developer.log(
+          record.message,
+          time: record.time,
+          name: record.loggerName,
+          level: record.level.value,
+          error: record.error,
+          stackTrace: record.stackTrace,
+        );
+
+        if (_loggersToShowInTerminal.contains(record.loggerName) && record.level >= Level.FINE) {
+          debugPrint('[${record.loggerName}] ${record.message}');
+        }
       }
+
+      _save(record);
+    });
+  }
+
+  void clear() {
+    state = (logs: const IList.empty());
+  }
+
+  void _save(LogRecord record) {
```

Comment:
Ok, let's not use a Provider here to avoid this.

I don't think we need to watch changes here, we just want to display the last snapshot. So you can convert the notifier to a service that can expose its current state through a method.